internal/connector/connector.go

```diff
@@ -343,10 +343,9 @@ func (c *CertCache) AddHost(host string) error {
 }
 
 func (c *CertCache) Certificate() (*tls.Certificate, error) {
-	c.mu.Lock()
-	defer c.mu.Unlock()
-
```

Comment:
the lock protects c.cert, so you can't check it's nil without the lock.

Reply:
In the case multiple requests find it to be nil it will be ok due to locks and checks in the AddHost function.

For example:
1. Request A comes in and finds that the cert is nil.
2. At the same time request B comes in and finds that the cert is nil.
3. They both call c.AddHost("")
4. The addHost function has a lock that one of the requests gets.
5. The request adds the cert and releases the lock.
6. When the other request attemps to re-add the cert for the same host it finds that the host already was added and returns without re-generating.

Reply:
multiple goroutines can't even call c.cert without being within a lock. it's not a matter of races.

Reply:
I see, thanks, I'll put a different lock here